src/main/java/seedu/address/model/appointment/AppointmentContainsDatePredicate.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.model.appointment;
+
+import static seedu.address.commons.util.DateUtil.DATE_FORMATTER;
+
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Appointment}'s start or end date matches the date given.
+ * Returns True if the date given is null and the person has an appointment.
+ */
+public class AppointmentContainsDatePredicate implements Predicate<Person> {
+    private final LocalDate date;
+
+    public AppointmentContainsDatePredicate(LocalDate date) {
+        this.date = date;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (person.getAppointment() == null) {
```

Comment:
If person is null, will it throw a null exception?

Reply:
I think if person is null, a null pointer exception will be thrown as you cannot call methods on null references. Consider adding an additional layer of checking to handle null cases?